app/src/components/SoldArtworkList.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import SoldArtworkItem from './SoldArtworkItem';
+import CardColumns from 'react-bootstrap/CardColumns';
+import { ContractProps } from '../helper/eth';
+
+interface SoldArtworkListState {
+  balance: number;
+  tokenIds: Array<number>;
+  prices: Array<number>;
+  newOwners: Array<string>;
+}
+
+class SoldArtworkList extends React.Component<ContractProps, SoldArtworkListState> {
+  constructor (props: ContractProps) {
+    super(props);
+    this.state = {
+      balance: 0,
+      tokenIds: [],
+      prices: [],
+      newOwners: [],
+    };
+  }
+
+  componentDidMount (): void {
+    this.loadSales();
+  }
+
+  shouldComponentUpdate (): boolean {
+    this.loadSales();
+    return true;
+  }
+
+  async loadSales (): Promise<void> {
+    const governance = this.props.contracts.Governance;
+    const currentAccount = this.props.accounts[0];
+    let eventsFound = 0;
+    const tokensArray: any[] = [];
+    const priceArray: any[] = [];
+    const newOwnerArray: any[] = [];
+
+    await governance.getPastEvents(
+      'RecordARR',
+      (errors: any, events: any) => {
+        if (!errors) {
+          for (let i = 0; i < events.length; i++) {
+            if (events[i].returnValues.from === currentAccount) {
+              eventsFound += 1;
+              tokensArray.push(events[i].returnValues.tokenId);
+              priceArray.push(events[i].returnValues.price);
+              newOwnerArray.push(events[i].returnValues.to);
```

Comment:
It would, I was going for the quicker easier solutions for things in general because I’m not super familiar with typescript but was thinking this at the time: will make this change!